src/curl.c

```diff
@@ -480,6 +480,8 @@ CURLcode swupd_curl_set_basic_options(CURL *curl, const char *url)
 		goto exit;
 	}
 
+	curl_easy_setopt(curl, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_0);
+
```

Comment:
While I see no problem with this,  I'd propose we let curl decide which to use with:
   CURL_HTTP_VERSION_NONE   From curl.haxx.se docs: "We don't care about what version the library uses. libcurl will use whatever it thinks fit.", 

or even 

      CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE : 
          "Issue non-TLS HTTP requests using HTTP/2 without HTTP/1.1 Upgrade. It requires prior knowledge that the server supports HTTP/2 straight away. HTTPS requests will still do HTTP/2 the standard way with negotiated protocol version in the TLS handshake. "